x/qgb/keeper/keeper_data_commitment.go

```diff
@@ -52,3 +63,23 @@ func (k Keeper) GetDataCommitmentForHeight(ctx sdk.Context, height uint64) (type
 	}
 	return types.DataCommitment{}, fmt.Errorf("data commitment for height not found")
 }
+
+// GetLastDataCommitment returns the last data commitment.
+func (k Keeper) GetLastDataCommitment(ctx sdk.Context) (types.DataCommitment, error) {
+	latestNonce := k.GetLatestAttestationNonce(ctx)
+	for i := uint64(0); i < latestNonce; i++ {
+		att, found, err := k.GetAttestationByNonce(ctx, latestNonce-i)
+		if err != nil {
+			return types.DataCommitment{}, err
+		}
+		if !found {
+			return types.DataCommitment{}, fmt.Errorf("couldn't find attestation with nonce %d", latestNonce-i)
```

Comment:
[non-blocking]
do we have an specific error type for this? it feels general enough to warrant one

Reply:
This we're creating errors on the fly throughout the whole package, I created an issue to handle the  all https://github.com/celestiaorg/celestia-app/issues/1390